lib/linter/linter.js

```diff
@@ -1608,25 +1636,109 @@ class Linter {
+
+                // need to verify rule existence options
+                if (inlineConfigResult.ok) {
+
+                    const ruleValidator = new RuleValidator();
 
+                    for (const { config: inlineConfig, node } of inlineConfigResult.configs) {
+
+                        Object.keys(inlineConfig.rules).forEach(ruleId => {
+                            const rule = getRuleFromConfig(ruleId, config);
+                            const ruleValue = inlineConfig.rules[ruleId];
+
+                            if (!rule) {
+                                inlineConfigProblems.push(createLintingProblem({ ruleId, loc: node.loc }));
+                                return;
+                            }
+
+                            try {
+
+                                const ruleOptions = Array.isArray(ruleValue) ? ruleValue : [ruleValue];
+
+                                assertIsRuleOptions(ruleId, ruleValue);
+                                assertIsRuleSeverity(ruleId, ruleOptions[0]);
+
+                                ruleValidator.validate({
+                                    plugins: config.plugins,
+                                    rules: {
+                                        [ruleId]: ruleOptions
+                                    }
+                                });
+                                mergedInlineConfig.rules[ruleId] = ruleValue;
+                            } catch (err) {
+
+                                let baseMessage = err.message.slice(
+                                    err.message.startsWith("Key \"rules\":")
+                                        ? err.message.indexOf(":", 12) + 1
+                                        : err.message.indexOf(":")
+                                ).trim();
+
+                                if (err.messageTemplate) {
+                                    baseMessage += ` You passed "${ruleValue}".`;
+                                }
+
+                                inlineConfigProblems.push(createLintingProblem({
+                                    ruleId,
+                                    message: `Inline configuration for rule "${ruleId}" is invalid:\n\t${baseMessage}\n`,
+                                    loc: node.loc
+                                }));
+                            }
+                        });
+                    }
+                }
+            }
+        }
+
+        const commentDirectives = options.allowInlineConfig && !options.warnInlineConfig
+            ? getDirectiveCommentsForFlatConfig(
+                sourceCode,
+                ruleId => getRuleFromConfig(ruleId, config)
+            )
+            : { configuredRules: {}, enabledGlobals: {}, exportedVariables: {}, problems: [], disableDirectives: [] };
+
+        const configuredRules = Object.assign({}, config.rules, mergedInlineConfig.rules, commentDirectives.configuredRules);
```

Comment:
```suggestion
            : { problems: [], disableDirectives: [] };

        const configuredRules = Object.assign({}, config.rules, mergedInlineConfig.rules);
```

These seem unnecessary now.